docs/pages/access-controls/guides/hardware-key-support.mdx

```diff
@@ -0,0 +1,215 @@
+  spec:
+    type: local
+    require_session_mfa: hardware_key_touch
+  ```
+
+  Update the configuration:
+
+  ```code
+  $ tctl create -f cap.yaml
+  # cluster auth preference has been updated
+  ```
+
+  </TabItem>
+</Tabs>
+</ScopedBlock>
+
+<ScopedBlock scope={["cloud"]}>
+
+Obtain your existing `cluster_auth_preference` resource:
+
+```code
+$ tctl get cap > cap.yaml
+```
+
+If you have not defined a `cluster_auth_preference`, `cap.yaml` will be blank.
+
+Ensure that `cap.yaml` includes the following content:
+
+```yaml
+kind: cluster_auth_preference
+version: v2
+metadata:
+  name: cluster-auth-preference
+spec:
+  type: local
+  require_session_mfa: hardware_key_touch
+```
+
+Update the configuration:
+
+```code
+$ tctl create -f cap.yaml
+# cluster auth preference has been updated
+```
+
+</ScopedBlock>
+
+## Step 2/2. Log in
+
+Once enabled on the Auth server, affected users will be prompted to connect and touch their YubiKey on login:
+
+<ScopedBlock scope={["oss", "enterprise"]}>
+
+```code
+$ tsh login --user=dev --proxy=proxy.example.com:3080
+# Enter password for Teleport user dev:
+# Tap your YubiKey
+# > Profile URL:        https://example.com
+#   Logged in as:       dev
+#   Cluster:            example.com
+#   Roles:              access, editor
+#   Logins:             bjoerger
+#   Kubernetes:         enabled
+#   Valid until:        2022-10-11 01:53:44 -0700 PDT [valid for 8h0m0s]
+#   Extensions:         permit-X11-forwarding, permit-agent-forwarding, permit-port-forwarding, permit-pty, private-key-policy
+```
+
+</ScopedBlock>
+
+<ScopedBlock scope={["cloud"]}>
+
+```code
+$ tsh login --proxy=mytenant.teleport.sh
+# Enter password for Teleport user dev:
+# Tap your YubiKey
+# > Profile URL:        https://example.com
+#   Logged in as:       dev
+#   Cluster:            example.com
+#   Roles:              access, editor
+#   Logins:             bjoerger
+#   Kubernetes:         enabled
+#   Valid until:        2022-10-11 01:53:44 -0700 PDT [valid for 8h0m0s]
+#   Extensions:         permit-X11-forwarding, permit-agent-forwarding, permit-port-forwarding, permit-pty, private-key-policy
+```
+
+</ScopedBlock>
+
+Affected user's with existing sessions will be prompted to re-login on their next request:
+
+```code
+$ tsh clusters
+# Unmet private key policy "hardware_key_touch"
+# Relogging in with YubiKey generated private key.
+# Enter password for Teleport user dev:
+# Tap your YubiKey
+# Cluster Name Status Cluster Type Labels Selected 
+# ----------- ------ ------------ ------ -------- 
+# example.com online root                *        
+```
+
+## Configuring Hardware Key support with per-session MFA
+
+For existing users of per-session MFA, upgrading to `hardware_key_touch` may be disruptive, as it requires touch on all requests as opposed to only session requests. While this upgrade would yield the best security, we can get a very similar level of security by combining Hardware Key support with per-session MFA.
+
```

Comment:
So we support both `hardware_key` and `hardware_key_touch` and they behave differently?

You have to "touch" a hardware key in both cases, right? I find the choice of names here pretty confusing.

Reply:
I agree, the confusion comes from the per-session MFA touch being completely separate from PIV touch. I think there's an argument for removing `hardware_key` altogether, and only having `hardware_key_touch` (probably changing the name to `hardware_key`), but it seems a bit late to do that.

Ultimately this is the solution we were able to agree on after some [back and forth in the RFD](https://github.com/gravitational/teleport/pull/14774#issuecomment-1218306104).

Reply:
Yes, we don't want users to enable MFA and PIV at the same time.

Reply:
Sure but is it too late to at least have better names? Both of them sound the same to me.

Reply:
For some background, the names originally were made for a new field: `private_key_policy: none | hardware_key | hardware_key_touch`, which as I mentioned is separate from per-session MFA, but was rejected in RFD review. In the context of the field `require_session_mfa`, neither of these make sense, and I don't know how to make them make sense. I'm really at a loss for any better names, but I'm open to suggestions.